plugins/processors/t128_filter/README.md

```diff
@@ -0,0 +1,74 @@
+# T128 Filter Processor Plugin
+
+The `t128_filter` filters out metrics passing through it. This processor is useful if the sort of filtering you wish to accomplish which is not possible with `tagpass`/`tagdrop`. The most usual case this processor can help is if you have two tags that you want both to be a certain value before passing along the metric.
+
+[Metric selectors](docs/CONFIGURATION.md#selectors) (such as `tagpass` and `tagdrop`) can be used to only apply the filters to specific metrics.
+
+### Configuration:
+
+```toml
+[[processors.t128_filter]]
+  ## The conditions that must be met to pass a metric through. This is similar
+  ## behavior to a tagpass, but multiple tags are ANDed
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+     #tag1 = ["value1", "value2"]
+     #tag2 = ["value3"]
+```
+
+### Basic Example Filter:
+
+Multiple values for a specific tag are OR'd together. A metric without a tag is also dropped.
+
+```toml
+[[processors.t128_filter]]
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+    tag1 = ["value1", "value2"]
+```
+
+```diff
+measurement tag1=value1 1612214810000000000
+measurement tag1=value2 1612214810000000000
+- measurement tag1=value3 1612214805000000000
+- measurement 1612214805000000000
+```
+
+### Multiple Keys Example Filter:
+
+Multiple keys are AND'd together.
+
+```toml
+[[processors.t128_filter]]
+  [[processors.t128_filter.condition]]
+
+  [processors.t128_filter.condition.tags]
+    tag1 = ["value1""]
+    tag2 - ["value2"]
+```
+
+```diff
+measurement tag1=value1,tag2=value2 1612214810000000000
+- measurement tag2=value2 1612214810000000000
+- measurement tag1=value1 1612214805000000000
+- measurement tag1=value1,tag2=value3 1612214805000000000
+```
+
+### Multiple Conditions Example Filter:
```

Comment:
There needs to be more context around multiple conditions - specifially the fact that its a OR operation between conditions.

Reply:
I thought the line below has that context. Also I have it as AND by default rn

Reply:
Maybe I'm misunderstanding but here's what my interpretation is and I dont think thats not clear from the documentation at the moment

```toml
[[processors.t128_filter]]
  [[processors.t128_filter.condition]]
   name = "condition1"
  [processors.t128_filter.condition.tags]
    tag1 = ["value1"]

  [[processors.t128_filter.condition]]
   name = "condition2"
  [processors.t128_filter.condition.tags]
    tag2 = ["value2"]
```


The logic above would be: condition1 OR condition2 where each of the subcomponent of `condition1` is AND by default. I think you had a good example of this in the FS where you express the "pseudo code" for this. I think its worth adding something like that in this document for clarity.

Reply:
I think the description and the diff section which are both standard to telegraf processors is sufficient. Let me know if that still isn't clear.

Reply:
There may be some confusing about what `filter` means. I don't know if that's happening here, but I keep needing to think twice whether a match of the filter means to include or exclude the metric. Maybe `t128_drop` would clear things up.

Reply:
I actually think if we were to change it, we should change it to t128_pass since whatever matches will get passed on, so it is more similar to tagpass

Reply:
This demonstrates the confusion. A pass is a simple negation of a drop. So we just need to use terminology that makes it clear. Either is possible if desired.

Reply:
I think we're overthinking it. The filter in this instance is actually correct IMO.